altanalyze3/components/merging/main_new.py

```diff
@@ -0,0 +1,75 @@
+from genericpath import exists
+from operator import truediv
+
+import os
+from functools import cache
+import logging
+
+class   JunctionAnnotation:
+    def __init__(self):
+        self.start_annotation = {},
+        self.stop_annotation = {},
+        self.junction_coordinate_BAM_dict = {}
+
+    def each_junction_annotation(self,junction_dir, gene_model_all):
+        '''
+        Input args: path to all BAM derived sample txt files, gene_model_all txt file(hg38)
+        Output: map of annotations for a given junction
+        '''
+        junction_files = os.listdir(junction_dir)
+        junction_dict = self.generate_junction_dict(gene_model_all)
+        junction_coordinate_BAM_listof_dicts = []
+        for junction_file in junction_files:
+            logging.info("Reading junction file" + '' + junction_file)
+            with open(junction_dir + junction_file) as f:
+                
+                for junction in f:
+                    chr = junction.split('\t')[0]
+                    junction_start = int(junction.split('\t')[1])
+                    junction_stop = int(junction.split('\t')[2])
+                    start_tar_tup = (chr, junction_start)
+                    stop_tar_tup = (chr, junction_stop)
+                    #print(start_tar_tup)
+                    if junction_dict.get(start_tar_tup) != None:
+                        
+                        self.start_annotation = {'exon_region_id':junction_dict[start_tar_tup]['exon_region_id'],
+                        'gene_id':junction_dict[start_tar_tup]['gene_id'],'junction_start':junction_start}
+                        #print(self.start_annotation)
+                    if junction_dict.get(stop_tar_tup) != None:
+                        self.stop_annotation = {'exon_region_id':junction_dict[stop_tar_tup]['exon_region_id'],
+                        'gene_id':junction_dict[stop_tar_tup]['gene_id'],'junction_stop':junction_stop}
+                    #print(self.start_annotation,self.stop_annotation)
+                    self.junction_coordinate_BAM_dict[(chr,junction_start,junction_stop)] = self.splice_annotations(self.start_annotation,self.stop_annotation)
+        #print(self.junction_coordinate_BAM_dict)
+                   
+        return self.junction_coordinate_BAM_dict
+
+   
+    @cache
```

Comment:
What's the reason for using cache here if this function is called only once?

Reply:
I am still experimenting with it.